src/common/controllers/ads-controller.js

```diff
@@ -228,9 +254,41 @@ class AdsController extends FakeEventTarget implements IAdsController {
     }
   }
 
+  _handlePrebidAdConfig(): void {
+    this._prebidManager &&
+      this._configAdBreaks
+        .filter(adBreak => !adBreak.played)
+        .map(adBreak => {
+          const loadPrebidAd = Promise.all(adBreak.ads.map(ad => this._getPrebidAds(ad)));
+          adBreak.loadedPromise = loadPrebidAd;
+          loadPrebidAd.then(ads => (adBreak.ads = ads));
+        });
+  }
+
+  _getPrebidAds(ad: KPAdObject): Promise<*> {
+    return new Promise(resolve => {
+      if (ad.prebid && this._prebidManager) {
+        const prebidConfig = Utils.Object.mergeDeep({}, ad.prebid, this._player.config.advertising.prebid);
+        const promiseLoad = this._prebidManager.load(prebidConfig);
+        promiseLoad
+          .then(bids => {
+            const vastUrls = bids.map(bid => bid && bid.vastUrl);
```

Comment:
if no bid.vastUrl we will get undefined as ad. need to filter before.

Reply:
It's not filtering, it's map I want the vastUrl so it could be also bid.vastUrl instead, it should return a list of vastUrls